### PR TITLE
Ensure graceful exit upon signals. Ignore SIGPIPE.

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Ensure that signals are handled such that:
1. `close(exit)` is called once, to prevent a panic.
2. The agent is given a chance to gracefully exit.
3. Handle `SIGPIPE` to prevent an abrupt exit upon `journald` crash.

### Motivation

Reports of system-probe exiting without proper cleanup of probes in #5462

### Additional Notes

`signal.Stop` will reset the signal behavior of the program to default Go behavior. This may cause an unexpected exit before proper cleanup is performed, if two signals come in rapid succession.

Returning from the function (exiting the goroutine) will guarantee we don't call `close(exit)` again, while allowing the agent to gracefully exit.

This `SIGPIPE` handling code was removed in #5200 but was serving a valuable purpose. If we do not handle the `SIGPIPE` signal, then the agent will be abruptly terminated if journald crashes.

### CODE REVIEW

File: pkg/process/util/signal_nowindows.go [hunk 1]
```diff
@@ -13,14 +13,20 @@ import (
 // HandleSignals tells us whether we should exit.
 func HandleSignals(exit chan struct{}) {
 	sigIn := make(chan os.Signal, 100)
-	signal.Notify(sigIn, syscall.SIGINT, syscall.SIGTERM)
+	signal.Notify(sigIn, syscall.SIGINT, syscall.SIGTERM, syscall.SIGPIPE)
 	// unix only in all likelihood; but we don't care.
 	for sig := range sigIn {
 		switch sig {
 		case syscall.SIGINT, syscall.SIGTERM:
-			signal.Stop(sigIn)
```
Comment (heyronhay) [inline pkg/process/util/signal_nowindows.go:21]: Why was this removed?

Comment (heyronhay) [reply pkg/process/util/signal_nowindows.go:21]: Nevermind, read the PR text a bit better ;)

**Review by heyronhay — APPROVED** [review]

**Review by leeavital — APPROVED** [review]
